### PR TITLE
Update URL API

### PR DESCRIPTION
**Pull request by Progitiel**
Move from deprecated Legacy API to WHATWG URL API.
[https://nodejs.org/api/url.html#url_legacy_url_api](url)

### CODE REVIEW

**Review by TimeForANinja — COMMENTED** [review]
Hey,
first of all thanks for the PR.
I'm fine with discussing adding support for RD-Playlists but for now i'd be happy keeping the two `.startsWith('RD')` statements out of this PR to merge it fast 😉

**Comment by Progitiel** [comment]
I agree. With this goal in mind, the two `.startsWith('RD')` don't need to be added.

File: lib/main.js [hunk 1]
```diff
@@ -144,22 +143,23 @@ main.validateID = linkOrId => {
     return false;
   }
   // Clean id provided
-  if (PLAYLIST_REGEX.test(linkOrId) || ALBUM_REGEX.test(linkOrId)) {
+  if (PLAYLIST_REGEX.test(linkOrId) || ALBUM_REGEX.test(linkOrId) || linkOrId.startsWith('RD')) {
```
Comment (TimeForANinja) [inline lib/main.js:146]:
```suggestion
  if (PLAYLIST_REGEX.test(linkOrId) || ALBUM_REGEX.test(linkOrId)) {
```

File: lib/main.js [hunk 2]
```diff
@@ -193,22 +193,23 @@ main.getPlaylistID = async linkOrId => {
     throw new Error('The linkOrId has to be a string');
   }
   // Clean id provided
-  if (PLAYLIST_REGEX.test(linkOrId) || ALBUM_REGEX.test(linkOrId)) {
+  if (PLAYLIST_REGEX.test(linkOrId) || ALBUM_REGEX.test(linkOrId) || linkOrId.startsWith('RD')) {
```
Comment (TimeForANinja) [inline lib/main.js:196]:
```suggestion
  if (PLAYLIST_REGEX.test(linkOrId) || ALBUM_REGEX.test(linkOrId)) {
```

**Review by TimeForANinja — CHANGES_REQUESTED** [review]
Hey
would be cool if we can get this done 😉

**Comment by TimeForANinja** [comment]
👍 Thanks again for this PR
gonna push to npm in a sec

**Pull request merged**